frmts/hdf5/bagdataset.cpp

```diff
@@ -1441,6 +1471,708 @@ CPLErr BAGResampledBand::IReadBlock(int nBlockXOff, int nBlockYOff,
+/*                           InitializeMinMax()                         */
+/************************************************************************/
+
+void BAGInterpolatedBand::InitializeMinMax()
+{
+    BAGDataset *poGDS = cpl::down_cast<BAGDataset *>(poDS);
+    if (nBand == 1 &&
+        GH5_FetchAttribute(poGDS->m_hVarresRefinements, "max_depth",
+                           m_dfMaximum) &&
+        GH5_FetchAttribute(poGDS->m_hVarresRefinements, "min_depth",
+                           m_dfMinimum))
+    {
+        m_bMinMaxSet = true;
+    }
+    else if (nBand == 2 &&
+             GH5_FetchAttribute(poGDS->m_hVarresRefinements, "max_uncrt",
+                                m_dfMaximum) &&
+             GH5_FetchAttribute(poGDS->m_hVarresRefinements, "min_uncrt",
+                                m_dfMinimum))
+    {
+        m_bMinMaxSet = true;
+    }
+}
+
+/************************************************************************/
+/*                             GetMinimum()                             */
+/************************************************************************/
+
+double BAGInterpolatedBand::GetMinimum(int *pbSuccess)
+
+{
+    if (m_bMinMaxSet)
+    {
+        if (pbSuccess)
+            *pbSuccess = TRUE;
+        return m_dfMinimum;
+    }
+
+    return GDALRasterBand::GetMinimum(pbSuccess);
+}
+
+/************************************************************************/
+/*                             GetMaximum()                             */
+/************************************************************************/
+
+double BAGInterpolatedBand::GetMaximum(int *pbSuccess)
+
+{
+    if (m_bMinMaxSet)
+    {
+        if (pbSuccess)
+            *pbSuccess = TRUE;
+        return m_dfMaximum;
+    }
+
+    return GDALRasterBand::GetMaximum(pbSuccess);
+}
+
+/************************************************************************/
+/*                     BarycentricInterpolation()                       */
+/************************************************************************/
+
+static bool BarycentricInterpolation(double dfX, double dfY,
+                                     const double adfX[3], const double adfY[3],
+                                     double &dfCoord0, double &dfCoord1,
+                                     double &dfCoord2)
+{
+    /* See https://en.wikipedia.org/wiki/Barycentric_coordinate_system */
+    double dfDenom = (adfY[1] - adfY[2]) * (adfX[0] - adfX[2]) +
+                     (adfX[2] - adfX[1]) * (adfY[0] - adfY[2]);
+    if (fabs(dfDenom) < 1e-5)
+    {
+        // Degenerate triangle
+        return false;
+    }
+    double dfMul0X = (adfY[1] - adfY[2]) / dfDenom;
+    double dfMul0Y = (adfX[2] - adfX[1]) / dfDenom;
+    double dfMul1X = (adfY[2] - adfY[0]) / dfDenom;
+    double dfMul1Y = (adfX[0] - adfX[2]) / dfDenom;
+    double dfCstX = adfX[2];
+    double dfCstY = adfY[2];
+    dfCoord0 = dfMul0X * (dfX - dfCstX) + dfMul0Y * (dfY - dfCstY);
```

Comment:
maybe pre-calculate (dfY - dfCstY) for efficiency.

Reply:
done